meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -236,6 +239,10 @@ public ConnectState getClientConnectionState() {
   @Override
   public boolean subscribeDataChange(String key, DataChangeListener listener,
       boolean skipWatchingNonExistNode, boolean persistListener) {
+    if (!persistListener) {
+      throw new NotImplementedException("Non-persist (one-time) listener is not supported in ZkMetaClient.");
```

Comment:
nit: should we add "currently" or we will never support it?

Reply:
good point, updated. thanks